common/repo/tests/GitRepoTest.java

```diff
@@ -156,5 +161,149 @@ public void testGetMultipleCommits() {
                 .build()),
         repo.getCommits(TEST_BRANCH));
   }
+
+  @Test
+  public void testAddFile() {
+    repo.switchBranch(TEST_BRANCH);
+    fileUtils.writeStringUnchecked(TEST_FILE_CONTENTS, fileUtils.joinPaths(repoFolder, TEST_FILE));
+    gitRepo.addFile(TEST_FILE);
+    repo.commit(repo.getUncommittedFiles(), COMMIT_MESSAGE);
+    assertEquals(0, repo.getUncommittedFiles().size());
+  }
+
+  @Test
+  public void testGetFileContents() {
+    repo.switchBranch(TEST_BRANCH);
+    fileUtils.writeStringUnchecked(TEST_FILE_CONTENTS, fileUtils.joinPaths(repoFolder, TEST_FILE));
+    gitRepo.addFile(TEST_FILE);
+    repo.commit(repo.getUncommittedFiles(), COMMIT_MESSAGE);
+    String commitId = gitRepo.getHeadCommitId();
+    assertEquals(TEST_FILE_CONTENTS, repo.getFileContents(commitId, TEST_FILE));
+  }
+
+  @Test
+  public void testGetUncommittedFilesWhenAddedFile() {
+    repo.switchBranch(TEST_BRANCH);
+    fileUtils.writeStringUnchecked(
+        TEST_FILE_CONTENTS, fileUtils.joinPaths(repoFolder, "added_file.txt"));
+    gitRepo.addFile("added_file.txt");
+    assertEquals(
+        ImmutableList.of(File.newBuilder().setFilename("added_file.txt").build()),
+        repo.getUncommittedFiles());
+  }
+
+  @Test
+  public void testGetUncommittedFilesWhenModifiedFile() {
+    repo.switchBranch(TEST_BRANCH);
+    fileUtils.writeStringUnchecked(TEST_FILE_CONTENTS, fileUtils.joinPaths(repoFolder, TEST_FILE));
+    gitRepo.addFile(TEST_FILE);
+    repo.commit(repo.getUncommittedFiles(), COMMIT_MESSAGE);
+    fileUtils.writeStringUnchecked("new file contains", fileUtils.joinPaths(repoFolder, TEST_FILE));
```

Comment:
You probably meant:
`s/new file contains/new file contents`

Reply:
Fixed.